### ISSUE TITLE
Questions about data augmentation

### ISSUE THREAD

**Issue by tangweibh** [SinaHonari/RCN#6]
Hi Sina,

You mentioned in the paper that online data augmentation are applied to the 9,000 MTFL train set using random scale, rotation, and translation jittering. However, since I am not very familiar with Theano, I failed to figure out how you did this in the code even I have gone through the codes carefully. 

In the 'get_mini_batch_train' function, there is a section named #jittering the train set#. But I cannot find the data augmentation codes here. So could you please tell me which function or lines implement the online data augmentation in the code? 

Thanks a lot!

Best,
Wei

**Comment by SinaHonari** [comment]
Hi Wei,

You might have already figured it out, but in case you haven't the jittering is happening in 
https://github.com/SinaHonari/RCN/blob/master/models/create_procs.py#L229

There, a seed is taken from the seed_queue and the data is jittered using that seed. The jittered data for a given seed is used for an epoch of training.

The part you are referring to in get_mini_batch_train method only fetches a new jittered training-data from the data_queue (for a new epoch, that is when the mini-batches of the previous epoch is finished), and sets that as the new training data.

Best,
Sina

**Comment by tangweibh** [comment]
Hi Sina,

Thanks for your so detailed response. I did realize the data augmentation is implemented in a separate process. Thanks again for your sharing your code!

Best,
Wei
